test/envoye2e/driver/stackdriver_test.go

```diff
@@ -0,0 +1,318 @@
+address:
+  socket_address:
+    address: 127.0.0.1
+    port_value: {{ .Vars.ClientPort }}
+filter_chains:
+- filters:
+  - name: envoy.http_connection_manager
+    config:
+      codec_type: AUTO
+      stat_prefix: client{{ .N }}
+      http_filters:
+      - name: envoy.filters.http.wasm
+        config:
+          config:
+            vm_config:
+              runtime: "envoy.wasm.runtime.null"
+              code:
+                inline_string: "envoy.wasm.metadata_exchange"
+            configuration: "test"
+      - name: envoy.filters.http.wasm
+        config:
+          config:
+            root_id: "stackdriver_outbound"
+            vm_config:
+              vm_id: "stackdriver_outbound"
+              runtime: "envoy.wasm.runtime.null"
+              code:
+                inline_string: "envoy.wasm.null.stackdriver"
+            configuration: >-
+              {
+                "testMonitoringEndpoint": "127.0.0.1:{{ .Vars.SDPort }}",
+                "testLoggingEndpoint": "127.0.0.1:{{ .Vars.SDPort }}",
+              }
+      - name: envoy.router
+      route_config:
+        name: client
+        virtual_hosts:
+        - name: client
+          domains: ["*"]
+          routes:
+          - match: { prefix: / }
+            route:
+              cluster: server
+              timeout: 0s
+`
+
+const StackdriverServerHTTPListener = `
+name: server{{ .N }}
+traffic_direction: INBOUND
+address:
+  socket_address:
+    address: 127.0.0.1
+    port_value: {{ .Vars.ServerPort }}
+filter_chains:
+- filters:
+  - name: envoy.http_connection_manager
+    config:
+      codec_type: AUTO
+      stat_prefix: server{{ .N }}
+      http_filters:
+      - name: envoy.filters.http.wasm
+        config:
+          config:
+            vm_config:
+              runtime: "envoy.wasm.runtime.null"
+              code:
+                inline_string: "envoy.wasm.metadata_exchange"
+            configuration: "test"
+      - name: envoy.filters.http.wasm
+        config:
+          config:
+            root_id: "stackdriver_inbound"
+            vm_config:
+              vm_id: "stackdriver_inbound"
+              runtime: "envoy.wasm.runtime.null"
+              code:
+                inline_string: "envoy.wasm.null.stackdriver"
+            configuration: >-
+              {
+                "testMonitoringEndpoint": "127.0.0.1:{{ .Vars.SDPort }}",
+                "testLoggingEndpoint": "127.0.0.1:{{ .Vars.SDPort }}",
+              }
+      - name: envoy.router
+      route_config:
+        name: server
+        virtual_hosts:
+        - name: server
+          domains: ["*"]
+          routes:
+          - match: { prefix: / }
+            route:
+              cluster: inbound|9080|http|server.default.svc.cluster.local
+              timeout: 0s
+`
+
+func TestStackDriverPayload(t *testing.T) {
+	t.Parallel()
+	ports := Counter(19010)
+	params := &Params{
+		Vars: map[string]string{
+			"ClientPort":     strconv.Itoa(ports()),
```

Comment:
nit: can we use ports.go here: https://github.com/istio/proxy/blob/master/test/envoye2e/env/ports.go?

Reply:
Yes, it's better with ports.go. Updated.